### ISSUE TITLE
JavaDoc for org.cactoos.scalar.AndWithIndex should be updated 

### ISSUE THREAD

**Issue by alex-semenyuk** [yegor256/cactoos#690]
JavaDoc example from `org.cactoos.scalar.AndWithIndex` 

 
```
new And(
       new IterableOf<>("Mary", "John", "William", "Napkin"),
       new BiFuncOf<>(
              (text, index) -> System.out.printf("Name #%d: %s\n", index, text),
              true
       )
).value();
```

doesn't compile with error:

`Error:(45, 9) java: no suitable constructor found for And(org.cactoos.iterable.IterableOf<java.lang.String>,org.cactoos.func.BiFuncOf<java.lang.Object,java.lang.Object,java.lang.Boolean>)`

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by yegor256** [comment]
@alex-semenyuk fixed in 23693d9 thanks!
